### DISCUSSION TITLE
Questions about native scrobbing API

### DISCUSSION THREAD

**Discussion by jcorporation** [krateng/maloja#364] [Q&A] [answered]
Hello,

I am the developer of [myMPD](https://github.com/jcorporation/myMPD) and I want to add support for scrobbling to the native Maloja API.

I have read https://github.com/krateng/maloja/blob/master/API.md and have some questions, because the documentation was unclear to me or I can't find the info.

- ~How the scrobbler should provide the API key?~: Found in an other issue, I must append `?key=...` to the uri. Have I overseen this in the documentation?
- What means List(String) as type for artists and albumartists? A String, a string or an array, a comma separated string?

Thanks for any help.

**Comment by duckcantcode** [comment, marked as answer]
I see the first question was already answered, so let me attempt the second one.

The thing about maloja's native API is that it supports JSON also, which most everyone uses. So you can also do, in JSON, `"artists":["a","b"]`.

**Reply by jcorporation** [reply]
Thanks!

**Reply by duckcantcode** [reply]
This is not tested, if you'd like I can try, but: you _may_ be able to use comma separated strings in the URL.

Otherwise, you can POST with a JSON payload.

No problem!

**Reply by jcorporation** [reply]
An json array for artists and albumartists is perfect!

Should following payload for the `/apis/mlj_1/newscrobble?key=...` work?

```
{
    artists = ["artist1", "artist2"],
    title = "Title",
    album = "Album",
    albumartists = ["albumartist1", "albumartist2"],
    time = startTime
}
```

**Reply by duckcantcode** [reply]
JSON doesn't use equals signs. It'd probably be formed like this:
```json
{
    "artists":["artist1", "artist2"],
    "title":"Title",
    "album":"Album",
    "albumartists":["albumartist1", "albumartist2"],
    "time":startTime
}
```

**Reply by jcorporation** [reply]
Oh sure, I converted the struct from lua by hand…

**Reply by duckcantcode** [reply]
No problem. Let me know once your client is finished and I can add it to [my almost exhaustive list of maloja clients](https://github.com/duckfromdiscord/awesome-maloja).

**Reply by jcorporation** [reply]
The integration is now finished.

**Reply by duckcantcode** [reply]
Can you link me the commit where it was added?

**Reply by jcorporation** [reply]
It is not added in the main repository, but in the scripts repository from which users import scripts.

https://github.com/jcorporation/mympd-scripts/commit/d3bd83ffcbf104a3c695391b14816ed3eb6c82f0

**Reply by duckcantcode** [reply]
Added!
